distributed/scheduler.py

```diff
@@ -654,6 +659,59 @@ def _to_dict_no_nest(self, *, exclude: Container[str] = ()) -> dict[str, Any]:
             members=True,
         )
 
+    # Replace all scheduler updates of ws.processing with this.
+    # Occupancy arg is merely there for drop in replacement
+    def add_to_processing(self, ts: TaskState) -> None:
+        tg = ts.group
+        self.task_groups_count[tg.name] += 1
+        self.processing.add(ts)
+        for dts in ts.dependencies:
+            if self not in dts.who_has:
+                self.need_replica(dts)
+
+    # This will effectively replace Scheduler.get_comm_cost
+    # Comm cost is always called in
+    # - move_task_request
+    # - _set_duration_estimate
+    #    - add_to_processing
+    #    - _reevaluate_occupancy_worker
+
+    # TODO: Scheduler.add_replica should call this
+    def need_replica(self, ts: TaskState) -> None:
+        if ts not in self.needs_what:
+            self.needs_what[ts] = 1
+            self._network_occ += ts.get_nbytes()
+        else:
+            self.needs_what[ts] += 1
+
+    def dont_need_replica(self, ts: TaskState) -> None:
+        if ts in self.needs_what:
+            self.needs_what[ts] -= 1
+            if not self.needs_what[ts]:
+                del self.needs_what[ts]
+                self._network_occ -= ts.get_nbytes()
+
```

Comment:
This is currently not really used but I intended it to be used for `Scheduler.{add|remove}_replica`